### DISCUSSION TITLE
Converting schema field names from snake_case

### DISCUSSION THREAD

**Discussion by janbuchar** [orval-labs/orval#218] [Q&A] [answered]
I have an API that returns objects whose fields have snake_case names. openapi-generator can convert those to camelCase for me when I use it to generate typescript interfaces. Can Orval do something similar?

**Comment by anymaniax** [comment, marked as answer]
Not at the moment, the only solution is to use a [mutator](https://orval.dev/reference/configuration/output#mutator) that transforms the data of your API and a [transformer](https://orval.dev/reference/configuration/input#transformer) to transform all your types in camelCase. But could be a good feature to add

**Reply by janbuchar** [reply]
Thanks! The solution you describe seems feasible, even though baking the conversion into the generated code might be more convenient.

**Reply by janbuchar** [reply]
I experimented with the suggested solution and found a possible pitfall. When the API produces free-form key-value structures (`type:object` with `additionalProperties`), the mutator will update these as well, which is certainly unexpected.

**Reply by jordan-rigo** [reply]
Iam having the same issue. Did you find a working solution for this?

**Reply by janbuchar** [reply]
Unfortunately, no. Except the old-fashioned openapi-generator, which lacks a lot of features, I haven't found this functionality in any  OpenAPI code generator. It's frustrating and I'd be very happy if you could share your results should you find something :slightly_smiling_face:

**Comment by devraj** [comment]
We have a similar situation our API built in `python` with `FastAPI` has `snake_case` function and variables names and they are converted to `camelCase` without needing orval do anything extra. 

`FastAPI` allow us to provide a method that can return the `operation_id`:

```python
def generate_operation_id(route: APIRoute) -> str:
    """
        With a little help from FastAPI docs
        https://bit.ly/3rXeAvH

        Globally use the path name as the operation id thus
        making things a lot more readable, note that this requires
        you name your functions really well.

        Read  more about this on the FastAPI docs
        https://shorturl.at/vwz03
    """
    return route.name
```
and then configuration `FastAPI` to use this function:
```python
app = FastAPI(
    generate_unique_id_function=generate_operation_id,
    ....
)
```

`FastAPI` is tightly coupled with `pyndatic`, which can be configured to translate these:

```python
def to_lower_camel(name: str) -> str:
    """
    Converts a snake_case string to lowerCamelCase
    """
    upper = "".join(word.capitalize() for word in name.split("_"))
    return upper[:1].lower() + upper[1:]


class AppBaseModel(BaseModel):
    """ Pydantic base model for applications

    This class is used to define the base model for all schema
    that we use in the Application, it configures pydantic to
    translate between camcelCase and snake_case for the JSON
    amongst other default settings.

    populate_by_name will allow pydantic to translate SQLAlchemy 
    results into serializable models, while being able to translate
    the aliases back to the original names when serializing to JSON.

    For a full set of options, see:
    https://pydantic-docs.helpmanual.io/usage/model_config/
    """
    model_config = ConfigDict(
        populate_by_name=True,
        alias_generator=to_lower_camel,
    )
```

The solution is specific to `python` and `fastapi` but hopefully of some use in finding a relevant path forward.

**Comment by jordanrigo** [comment]
Do you guys plan on implementing this?
